tests/test_fetcher.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+
+"""
+<Program>
+  test_download.py
```

Comment:
Nit:
```suggestion
  test_fetcher.py
```

Maybe we should be explicit and call this test_requests_fetcher?

Reply:
Fixed the docstring but test_requests_fetcher sounds too verbose :)